### PR TITLE
Ref/bug/save as

### PR DESCRIPTION
**Pull request by MichiGandi**
Fix the scene id problem on the save as dialog.

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m92tkm fix: save as does save duplicate guid](https://app.clickup.com/t/861m92tkm)

File: Assets/Scripts/EditorState/SceneManagerState.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System.Linq;
 using JetBrains.Annotations;
 using Assets.Scripts.Utility;
+using System.Data;
```
Comment (cblech) [inline Assets/Scripts/EditorState/SceneManagerState.cs]: Using directive is unnecessary

File: Assets/Scripts/System/SceneManagerSystem.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System;
 using System.IO;
 using System.Linq;
+using UnityEditor;
```
Comment (cblech) [inline Assets/Scripts/System/SceneManagerSystem.cs]: Using directive is unnecessary
And will throw errors when building standalone

File: Assets/Scripts/System/SceneManagerSystem.cs [hunk 3]
```diff
@@ -243,19 +244,23 @@ private void SaveSceneAs(SceneDirectoryState sceneDirectoryState)
                 return;
             }
 
-            //remove any potential scene that will be overridden
+            // Strip path inside a dcl scene folder. This allows to save as a already existing scene.
+            newPath = newPath.Substring(0, newPath.IndexOf(".dclscene") + 9);
+
+            Guid newId = default;
```
Comment (cblech) [inline Assets/Scripts/System/SceneManagerSystem.cs]: Use `Guid.Empty` to be more explicit

**Review by cblech — CHANGES_REQUESTED** [review]

**Review by cblech — CHANGES_REQUESTED** [review]
There is a different problem now. When "Saving as..." this scene to a new location
grafik
I get this scene as a result
grafik

**Review by cblech — APPROVED** [review]

**Pull request merged**